utils/prod/update_config.py

```diff
@@ -29,48 +30,69 @@
 ]
 
 
+def get_nested_dict(data: Dict[str, Any], keys: List[str]) -> Optional[Dict[str, Any]]:
+    """Retrieve a nested dictionary given a list of keys."""
+    for key in keys:
+        if isinstance(data, dict) and key in data:
+            data = data[key]
+        else:
+            return None
+    return data if isinstance(data, dict) else None
+
+
+def update_nested_dict(data: Dict[str, Any], keys: List[str], value: Any) -> None:
+    """Update a nested dictionary given a list of keys and a value."""
+    for key in keys[:-1]:
+        data = data.setdefault(key, {})
+    data[keys[-1]] = value
```

Comment:
Ok so this is what happens to the last key. Why?
We need comments here to understand.

Reply:
The update_nested_dict function iterates through all keys except the last one to navigate to the appropriate nested dictionary, creating intermediate dictionaries if they don't exist (setdefault). Then, it updates the value of the last key.  Comments added